cpp/src/slice2cs/Gen.cpp

```diff
@@ -103,29 +103,65 @@ namespace
     }
 
     /// Returns a C# formatted link to the provided Slice identifier.
-    /// TODO: this is temporary and will be replaced when we add 'cs:identifier' support.
-    string csLinkFormatter(const string& rawLink, const ContainedPtr&, const SyntaxTreeBasePtr&)
+    string csLinkFormatter(const string& rawLink, const ContainedPtr& source, const SyntaxTreeBasePtr& target)
     {
-        string result = "<see cref=\"";
+        ostringstream result;
+        result << "<see ";
 
-        auto hashPos = rawLink.find('#');
-        if (hashPos != string::npos)
+        if (target)
         {
-            if (hashPos != 0)
+            if (auto builtinTarget = dynamic_pointer_cast<Builtin>(target))
             {
-                result += rawLink.substr(0, hashPos);
-                result += ".";
+                string typeS = CsGenerator::typeToString(builtinTarget, "");
+                if (typeS.find("Ice") == std::string::npos)
+                {
+                    result << "langword=\"" << typeS << "\"";
+                }
+                else
+                {
+                    // Remove trailing '?':
+                    typeS.pop_back();
+                    result << "cref=\"" << typeS << "\"";
+                }
+            }
+            else
+            {
+                // TODO: remove once mappedScope() is fixed.
+                string sourceScope = source->mappedScope(".").substr(1);
+                sourceScope.pop_back();
+
+                if (auto operationTarget = dynamic_pointer_cast<Operation>(target))
+                {
+                    // link to the method on the proxy interface
+                    result << "cref=\"" << CsGenerator::getUnqualified(operationTarget->interface(), sourceScope)
```

Comment:
This can generate fully qualified when partial qualified would work, but I couldn't find a good way to fix it.